linkerd/app/core/src/proxy/server.rs

```diff
@@ -163,8 +166,15 @@ where
                     .forward_tcp
                     .clone()
                     .into_service()
-                    .oneshot((proto.tls, io));
-                return Box::new(drain.watch(fwd.map_err(Into::into), |_| {}));
+                    .oneshot((proto.tls, io))
+                    .map(|connect_future| {
+                        let connect_future = drain
+                            .watch(connect_future.map_err(Into::into), |_| {})
+                            .map_err(Into::into);
+                        Box::new(connect_future) as Self::Response
```

Comment:
Little surprised this cast is necessary --- have you tried removing it? Does the compiler complain?

Reply:
@hawkw I have, it errors out with: 

```
177 |return Box::new(fwd.map_err(Into::into));
    |       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected trait `futures::future::Future`, found struct `futures::future::map_err::MapErr`
```